### PR TITLE
Create a class for retrieving and verifying OCR fields

### PR DESCRIPTION
**Pull request by lgonczar**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/BPS-570

### Change description ###

Create a class for retrieving and verifying OCR fields. Also, fix `ResultOrErrors` to handle null results. 

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/test/java/uk/gov/hmcts/reform/bulkscanccdeventhandler/services/OcrFieldRetrieverTest.java [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+import uk.gov.hmcts.reform.bulkscanccdeventhandler.model.ResultOrErrors;
+
+import java.time.LocalDate;
+import java.util.Arrays;
+import java.util.stream.Stream;
+
+import static java.util.stream.Collectors.toList;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.Assert.assertTrue;
+import static org.junit.jupiter.api.Assertions.assertFalse;
```
Comment (rkondratowicz) [inline src/test/java/uk/gov/hmcts/reform/bulkscanccdeventhandler/services/OcrFieldRetrieverTest.java]: no need for those if you're using assertJ already

Comment (lgonczar) [reply src/test/java/uk/gov/hmcts/reform/bulkscanccdeventhandler/services/OcrFieldRetrieverTest.java]: Oh great. What a mix. Will fix.

File: src/main/java/uk/gov/hmcts/reform/bulkscanccdeventhandler/services/OcrFieldRetriever.java [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+            if (isNullOrEmpty(stringValue)) {
+                return isRequired
+                    ? errors(String.format("OCR field %s must have a value", fieldName))
+                    : result(null);
+            } else {
+                return convertFieldValue(fieldName, stringValue, fieldClass);
+            }
+        } else {
+            return errors(String.format("There must be only one OCR field named %s", fieldName));
```
Comment (nitinprabhu) [inline src/main/java/uk/gov/hmcts/reform/bulkscanccdeventhandler/services/OcrFieldRetriever.java]: This scenario is not covered by test ?

File: src/test/java/uk/gov/hmcts/reform/bulkscanccdeventhandler/services/OcrFieldRetrieverTest.java [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+    }
+
+    @Test
+    public void should_return_error_when_field_required_but_not_found() {
+        OcrData ocrData = ocrData(new OcrDataField("field_1", "value"));
+
+        ResultOrErrors<String> result = ocrFieldRetriever(ocrData).getFieldValue(
+            "non-existing-field",
+            true,
+            String.class
```
Comment (nitinprabhu) [inline src/test/java/uk/gov/hmcts/reform/bulkscanccdeventhandler/services/OcrFieldRetrieverTest.java]: Would be nice to show this for Boolean and Date class too. I am not too strong on this but will leave it upto you :)

File: src/main/java/uk/gov/hmcts/reform/bulkscanccdeventhandler/services/OcrFieldRetriever.java (hunk 2 above)
Comment (lgonczar) [reply src/main/java/uk/gov/hmcts/reform/bulkscanccdeventhandler/services/OcrFieldRetriever.java]: as discussed in person, we've got a test for this case

File: src/test/java/uk/gov/hmcts/reform/bulkscanccdeventhandler/services/OcrFieldRetrieverTest.java (hunk 3 above)
Comment (lgonczar) [reply src/test/java/uk/gov/hmcts/reform/bulkscanccdeventhandler/services/OcrFieldRetrieverTest.java]: I'll cover primitive types

Comment (lgonczar) [reply src/test/java/uk/gov/hmcts/reform/bulkscanccdeventhandler/services/OcrFieldRetrieverTest.java]: Done for Integer, Boolean, LocalDate and String.

File: src/test/java/uk/gov/hmcts/reform/bulkscanccdeventhandler/services/OcrFieldRetrieverTest.java (hunk 1 above)
Comment (lgonczar) [reply src/test/java/uk/gov/hmcts/reform/bulkscanccdeventhandler/services/OcrFieldRetrieverTest.java]: Fixed

**Pull request merged**